File: Year-2023/LeetCode-February-Challenge/MaximumDepthofBinaryTree.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
class Solution {
    public int maxDepth(TreeNode root) {
        return root == null ? 0: 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
    }
}
public class MaximumDepthofBinaryTree{
    public static void main(String[] args) {
        TreeNode p1 = new TreeNode(3);
        TreeNode p2 = new TreeNode(9);
        TreeNode p3 = new TreeNode(20);
        TreeNode p4 = new TreeNode(15);
        TreeNode p5 = new TreeNode(7);

        p1.left = p2;
        p1.right = p3;
        p3.left = p4;
        p3.right = p5;
        
        Solution s = new Solution();
        System.out.println(s.maxDepth(p1));
    }
}

File: Year-2023/LeetCode-February-Challenge/MaximumDepthofBinaryTree.js
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}

var maxDepth = function(root) {
    return root === null ? 0: 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
};

let p1 = new TreeNode(3);
let p2 = new TreeNode(9);
let p3 = new TreeNode(20);
let p4 = new TreeNode(15);
let p5 = new TreeNode(7);

p1.left = p2;
p1.right = p3;
p3.left = p4;
p3.right = p5;

console.log(maxDepth(p1));